### PR TITLE
Add deps to --info output.

### PR DESCRIPTION
**Pull request by ry**
Move module stuff into its own file.
cc @kevinkassimo

### CODE REVIEW

File: src/modules.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      Deps {
+        name,
+        depth,
+        deps: None,
+      }
+    } else {
+      seen.insert(id);
+      let child_ids = modules.get_children(id).unwrap();
+      let deps = child_ids
+        .into_iter()
```
Comment (piscisaureus) [inline src/modules.rs]: this .into_iter() call is equivalent to .iter() and will not move the Vec
note: #[warn(clippy::into_iter_on_ref)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#into_iter_on_ref
help: call directly: `iter`

File: src/modules.rs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+impl Modules {
+  pub fn new() -> Modules {
+    Self {
+      info: HashMap::new(),
+      by_name: HashMap::new(),
+    }
+  }
+
+  pub fn get_id(&self, name: &str) -> Option<deno_mod> {
+    self.by_name.get(name).map(|id_ref| *id_ref)
```
Comment (piscisaureus) [inline src/modules.rs]: You are using an explicit closure for cloning elements
note: #[warn(clippy::map_clone)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
help: Consider calling the dedicated `cloned` method: `self.by_name.get(name).cloned()`

File: src/modules.rs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+/// A collection of JS modules.
+pub struct Modules {
+  pub info: HashMap<deno_mod, ModuleInfo>,
+  pub by_name: HashMap<String, deno_mod>,
+}
+
+impl Modules {
+  pub fn new() -> Modules {
```
Comment (piscisaureus) [inline src/modules.rs]: you should consider deriving a `Default` implementation for `modules::Modules`

note: #[warn(clippy::new_without_default)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default

File: src/modules.rs (hunk 1 above)
Comment (ry) [reply src/modules.rs]: Fixed.

File: src/modules.rs (hunk 3 above)
Comment (ry) [reply src/modules.rs]: Fixed.

File: src/modules.rs (hunk 2 above)
Comment (ry) [reply src/modules.rs]: Fixed.

File: src/modules.rs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        deps: Some(deps),
+      }
+    }
+  }
+}
+
+impl fmt::Display for Deps {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    for _i in 0..self.depth {
+      write!(f, "| ")?;
```
Comment (piscisaureus) [inline src/modules.rs]: This bar | is not helpful IMO.

Comment (ry) [reply src/modules.rs]: Very open to reformatting

Comment (piscisaureus) [reply src/modules.rs]: Barebones:

```
root
  aaa
    alpha
      first
      second
    beta
      foo
      bar
  bbb
    something
```

Basic:

```
root
- aaa
  * alpha
    - first
    - second
  * bbb
    - foo
    - bar
- bbb
  * something
```

Advanced:

```
root
├─ aaa
│  ├─ alpha
│  │  ├─ first
│  │  └─ second
│  └─ bbb
│     ├─ foo
│     └─ bar
└─ bbb
   └─ something
```

Comment (ry) [reply src/modules.rs]: advanced looks fine

**Pull request merged**